tab_mur_data_prep.Rmd

```diff
@@ -0,0 +1,167 @@
+---
+title: "Filter Cutoff Investigation"
+author: "C. Savonen, CCDL for ALSF"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+Created: 3/6/2019
+
+#### Purpose: Set up the tabula muris 10X dataset we are working with for normalization and subsequent steps. 
+
+Subgoal: The cutoffs used for the smart-seq2 single cell data do not work well 
+on the 10X single cell data. This is not surprising, but let's investigate the 
+10X tab mur data so we can come up with some better cutoffs. 
+
+Other note: For the purposes of saving room and making a more manageable dataset 
+to work with, I am only using P4, and have excluded the Liver and spleen samples. 
+
+```{r}
+# Load ggplot2 library
+library(ggplot2)
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Set up the data
+
+```{r}
+# Read in data
+all.data <- readRDS(file.path("tab_mur_data", "counts_tab_mur.RDS"))
+
+# Keep the gene names separate
+gene.names <- all.data$genes
+
+# Get rid of gene column
+all.data <- all.data[, -1]
+```
+ 
+Make cell info dataframe for labeling purposes
+
+```{r}
+# Clean up sample names
+colnames(all.data) <- gsub("_possorted_genome.bam", "", colnames(all.data))
+
+# Make a dataframe of this cell info
+cell.info <- data.frame(strsplit(colnames(all.data), "\\.")) %>% 
+  t() 
+
+# Change column names
+colnames(cell.info) <- c("cell_barcode", "sample.id")
+
+# Keep the full id names for each as well for matching purposes
+cell.info <- data.frame(full.id = colnames(all.data), cell.info)
+```
+
+Read in the metadata file obtained from figshare, clean it up so we can merge it
+with the sample names obtained in the cell.info dataframe. 
+
+```{r}
+# Import metadata file and make it with only the variables and samples we want
+metadata <- readr::read_csv(file.path("tab_mur_data", "metadata_droplet.csv")) %>% 
+  dplyr::select("channel", "mouse.id", "tissue") %>% 
+  dplyr::filter(grepl("10X_P4", channel), tissue != "Liver", tissue != "Spleen") 
+
+# Make column name matching so we can use inner_join to merge the info
+colnames(metadata)[1] <- "sample.id"
```

Comment:
In the future, you don't need the column names to match to use `inner_join`. You could use `by = c("first_df_colname" = "second_df_colname")` (see [this SO post](https://stackoverflow.com/questions/21888910/how-to-specify-names-of-columns-for-x-and-y-when-joining-in-dplyr)).